dart/lib/src/scope.dart

```diff
@@ -53,7 +53,13 @@ class Scope {
 
   Map<String, dynamic> get extra => Map.unmodifiable(_extra);
 
-  // TODO: Contexts
+  final Contexts _contexts = Contexts();
+
+  Contexts get contexts => _contexts;
+
+  void setContexts({String key, dynamic value}) {
```

Comment:
does it need to be named? because the other add methods are not, only this one will be different

Reply:
missing method `removeContexts`

Reply:
also, `setContexts` can be `dynamic` as value, but our protocol does not accept non objects here, see what is done on Java, we'd need something similar or `setContexts('a', 'b')` would cause a backend issue

https://github.com/getsentry/sentry-java/blob/main/sentry/src/main/java/io/sentry/Scope.java#L340-L374

Reply:
- `setContexts` args are now mandatory 
- added removeContexts
- I added some conditions to the setter ( not null and of type `num`, `bool` or `String` )
 What would be the best strategy in case of invalid value? For now, it's a no_op.

Reply:
Also, now, the value must be an Object ( with a `toString()` method ) so we could just set the toString() if the value is not a num, a bool, or a String?
```dart
if (key != null && value != null) return;

    _contexts[key] = (value is num || value is bool || value is String) ? value : '$value';
```

Reply:
@rxlabz on Java, we'd not return, but rather wrap into a `value` field if they are primitive types.
This deviates quite a lot of the Java impl. and limit the API quite a lot, I'd suggest doing something similar to Java.

exposing more setContexts methods or increment the current solution.

_contexts[key] = (value is num || value is bool || value is String) ? [{'value': value] : value;

sounds more realistic

Reply:
sorry, I missed the java `public void setContexts(final @NotNull String key, final @NotNull Object value)` and misunderstood the goal.

Reply:
✅